python/lsst/ip/isr/ampOffset.py

```diff
@@ -18,88 +18,433 @@
+        ampSides = np.full_like(ampAssociations, -1)
+
+        for ampId in ampIds.ravel():
+            neighbors, sides = self.getNeighbors(ampIds, ampId)
+            ampAssociations[ampId, neighbors] = -1
+            ampSides[ampId, neighbors] = sides
+            ampAssociations[ampId, ampId] = -ampAssociations[ampId].sum()
+
+        if ampAssociations.sum() != 0:
+            raise RuntimeError("The `ampAssociations` array does not sum to zero.")
+
+        self.log.debug("amp associations:\n%s", ampAssociations)
+        self.log.debug("amp sides:\n%s", ampSides)
+
+        return ampAssociations, ampSides
+
+    def getNeighbors(self, ampIds, ampId):
+        """Get the neighbor amplifiers and their sides for a given
+        amplifier.
+
+        Parameters
+        ----------
+        ampIds : `numpy.ndarray`
+            Matrix with amp side association information.
+        ampId : `int`
+            The amplifier ID for which neighbor amplifiers and side IDs
+            are to be found.
+
+        Returns
+        -------
+        neighbors : `list` [`int`]
+            List of neighbor amplifier IDs.
+        sides : `list` [`int`]
+            List of side IDs, with each ID corresponding to its respective
+            neighbor amplifier.
+        """
+        m, n = ampIds.shape
+        r, c = np.ravel(np.where(ampIds == ampId))
+        neighbors, sides = [], []
+        sideLookup = {
+            0: (r + 1, c),
+            1: (r, c + 1),
+            2: (r - 1, c),
+            3: (r, c - 1),
+        }
+        for side, (row, column) in sideLookup.items():
+            if 0 <= row < m and 0 <= column < n:
+                neighbors.append(ampIds[row][column])
+                sides.append(side)
+        return neighbors, sides
+
+    def getInterfaceOffset(self, ampIdA, ampIdB, edgeA, edgeB):
+        """Calculate the amp offset for a given interface between two
+        amplifiers.
 
         Parameters
         ----------
-        exposure : `lsst.afw.image.Exposure`
-            Exposure to be corrected for any amp offsets.
+        ampIdA : int
+            ID of the first amplifier.
+        ampIdB : int
+            ID of the second amplifier.
+        edgeA : numpy.ndarray
+            Amp edge for the first amplifier.
+        edgeB : numpy.ndarray
+            Amp edge for the second amplifier.
+
+        Returns
+        -------
+        interfaceOffset : float
+            The calculated amp offset value for the given interface between
+            amps A and B.
+        """
+        sctrl = StatisticsControl()
+        # NOTE: Taking the difference with the order below fixes the sign flip
+        # in the B matrix.
+        edgeDiff = edgeA - edgeB
+        if self.config.ampEdgeWindow == 0:
+            if len(edgeDiff) > self.shortAmpSide:
+                window = self.shortAmpSide
+            else:
+                window = int(self.shortAmpSide**2 / self.longAmpSide)
```

Comment:
I'm sure I'm missing something, but can you explain this choice?  This looks like it's triggered when the side under consideration is parallel to the short axis of the amplifier.  Why should the window used for that be shorter the longer the other axis is?

Reply:
We went with this method to keep things in line with the way it was done before in `obs_subaru`. The main purpose was to make sure the pixel size of the sliding window matches up with what's used in `obs_subaru`, so it works well with the HSC data we've already got. Now, I've simplified it further by replacing `ampEdgeWindow` with `ampEdgeWindowFrac`. With this new config option, we can easily say how much of the amp edge lengths we want as our sliding window for those rolling averages. The default is set to 512/4176, which matches the previous HSC setup.

Reply:
`self.longAmpSide` has been removed as a result of introducing `ampEdgeWindowFrac`.